### PR TITLE
New hats and caps

### PR DESCRIPTION
**Pull request by The-Anjelo-Guy**
#### Summary
Content "Adds several new hats, variants and their spawn rates."

#### Purpose of change

Just for having a little more content regarding hats.

### CODE REVIEW

**Comment by carlarctg** [comment]
You gave a bunch of hats the sunglasses flag and didn't mention that or why. Also these don't seem to be spawned anywhere.

**Comment by The-Anjelo-Guy** [comment]
> You gave a bunch of hats the sunglasses flag and didn't mention that or why. Also these don't seem to be spawned anywhere.

Still a work in progress. Also hats with sufficient long or wide brims such as the tophat should protect the eyes against the sun.

**Comment by fairyarmadillo** [comment]
I think the ten gallon hat may not be a good item to base the homburg, trilby etc. off of, as it is absolutely huge:

image

I would instead recommend the porkpie hat that we already have for the trilby, and the fedora that we already have for the homburg. This means trilbys won't have SUNGLASSES but will have FANCY, and homburgs will get both, which seems more in line with the existing hats that we have.

**Comment by The-Anjelo-Guy** [comment]
> Fun stuff! I think the ten gallon hat may not be a good item to base the homburg, trilby etc. off of, as it is absolutely huge:
> 
> image
> 
> I would instead recommend the porkpie hat that we already have for the trilby, and the fedora that we already have for the homburg. This means trilbys won't have SUNGLASSES but will have FANCY, and homburgs will get both, which seems more in line with the existing hats that we have.
> 
> The tyrolean hat should probably have neither, as it's neither a black tie appropriate item nor does it have a wide brim. You could perhaps copy-from the porkpie hat and then add "delete": { [ "FANCY" ] }
> 
> In the metal scrap cap, you have a grammar error:
> 
> > It is not much beautiful,
> 
> should be "It is not exactly beautiful", like you've done in the plastic cap.
> 
> The scrap caps should not have repairs_like a balaclava, because they incorporate plastic and metal and balaclavas don't. Lastly, you may want to add the plastic working and principles of metalworking proficiencies to either item, as other similar items (armored leather vest, for example) have things like that, and it'll be a good way for characters to get some early-game proficiency training.

Thank you so much for these suggestions, I appreciate it. I'll be applying them in no time!

File: data/json/items/armor/hats.json [hunk 1]
```diff
@@ -321,6 +461,360 @@
...
+      },
+      {
+        "id": "badminton_fan_ball",
+        "name": { "str": "badminton fan trucker cap" },
+        "description": "A snapback trucker cap displaying the logo of a popular badminton team.  It shields your eyes from the sun and provides a little bit of warmth.",
+        "weight": 1
+      },
+      {
+        "id": "flatball_fan_ball",
+        "name": { "str": "flatball fan trucker cap" },
```
Comment (Qrox) [inline data/json/items/armor/hats.json:690]: May I ask what is a flatball? I was trying to translate this but my search only turned up either 1. a type of strike in golf which I think can hardly be called its own sport, or 2. fla`g`ball which is a type of football.

Comment (The-Anjelo-Guy) [reply data/json/items/armor/hats.json:690]: Flatball is a type of table sport, kind of a kid's game but there are local clubs and surprisingly even teams of it. Flatball is not a ball, it is the name of the sport.

Comment (fairyarmadillo) [reply data/json/items/armor/hats.json:690]: In the US we usually call it disc golf. I think there's already disc golf stuff in the game.

Comment (Qrox) [reply data/json/items/armor/hats.json:690]: Thanks, but sadly I cannot find any reference on that so would you mind find one for me?

So far I've only found this (https://en.wikipedia.org/wiki/Ultimate_(sport)) which is nicknamed flatball but is obviously not a table sport.

Comment (The-Anjelo-Guy) [reply data/json/items/armor/hats.json:690]: > Thanks, but sadly I cannot find any reference on that so would you mind find one for me?
> 
> So far I've only found this (https://en.wikipedia.org/wiki/Ultimate_(sport)) which is nicknamed flatball but is obviously not a table sport.

Hey Qrox, I'm sorry but I misunderstood it. I remade my research (the content of this PR is some years old, I just updated and PRed it) and apparently you are right, the cap is actually a Ultimate (foosball) cap, I'm sorry.

**Pull request merged**